### ISSUE TITLE
UnusedImport reported for import org.gradle.kotlin.dsl.assign in kotlin-dsl

### ISSUE THREAD

**Issue by PhilippNowak96** [detekt/detekt#7360]
## Expected Behavior

No report

## Observed Behavior

Unused import is reported

## Steps to Reproduce

Add the following test to `UnusedImportSpec.kt` and see it failing:

```kotlin
@Test
fun `does not report equals operators in kotlin dsl`() {
    val main = """
        import org.gradle.api.Project
        import org.gradle.kotlin.dsl.assign
        import org.gradle.kotlin.dsl.configure
        import org.jetbrains.kotlin.gradle.dsl.JvmTarget
        import org.jetbrains.kotlin.gradle.dsl.KotlinJvmProjectExtension
        
        fun Project.configureKotlinJvm() {
            this.extensions.configure<KotlinJvmProjectExtension> {
                compilerOptions {
                    jvmTarget = JvmTarget.JVM_17
                }
            }
        }
    """.trimIndent()
    assertThat(subject.lintWithContext(env, main)).isEmpty()
}
```

## Context

Using `kotlin-dsl` in a `build-logic` module to for example configure the `jvmTarget` you need the `org.gradle.kotlin.dsl.assign` import to do things like `jvmTarget = JvmTarget.JVM_17` because the `=` operator resolves to the `assign` function.

## Your Environment

* Version of detekt used: 1.23.6

**Comment by ZhaohuaZeng-at** [comment]
+1 see the same issue
